tests/test_wrapper_writer.py

```diff
@@ -0,0 +1,189 @@
+
+    def test_read_structure_config(self):
+        w = WrapperWriter(self.method_config, self.structure_config)
+        w.read_configs()
+        config = w.structures
+
+        self.assertEqual(self.structure, config)
+
+    def test_no_structure_in_config(self):
+        w = WrapperWriter(self.method_config, self.method_config)
+        with self.assertRaises(Exception):
+            w.read_configs()
+
+    def test_instantiate_structure_class(self):
+        w = WrapperWriter(self.method_config, self.method_config)
+        w.structures = self.structure
+        w.instantiate_structure_class()
+        self.assertEqual("/test_dir/python/path/", w.structure_classes[0].path)
+        self.assertEqual("python.txt", w.structure_classes[0].template)
+        self.assertEqual("{}.py", w.structure_classes[0].file_name_format)
+
+        self.assertEqual("/test_dir/scala/path/", w.structure_classes[1].path)
+        self.assertEqual("scala.txt", w.structure_classes[1].template)
+        self.assertEqual("{}.scala", w.structure_classes[1].file_name_format)
+
+
+    def test_instantiate_container_class(self):
+        w = WrapperWriter(self.method_config, self.method_config)
+        w.containers = self.container
+        w.instantiate_container_class()
+
+        self.assertEqual("Maths", w.container_classes[0].name)
+        self.assertEqual("sum_column", w.container_classes[0].methods[0].name)
+        self.assertEqual({"column_a": "String", "column_b": "String"}, w.container_classes[0].methods[0].params)
+        self.assertEqual("This function adds the two columns together", w.container_classes[0].methods[0].docs)
+        self.assertEqual("String", w.container_classes[0].methods[0].returns)
+
+        self.assertEqual("mulitply", w.container_classes[0].methods[1].name)
+        self.assertEqual({"column_a": "String", "column_b": "String"}, w.container_classes[0].methods[1].params)
+        self.assertEqual("This function multiplies the two columns together", w.container_classes[0].methods[1].docs)
+        self.assertEqual("String", w.container_classes[0].methods[1].returns)
+
+        self.assertEqual("Estimation", w.container_classes[1].name)
+        self.assertEqual("ratio", w.container_classes[1].methods[0].name)
+        self.assertEqual({"column_a": "String", "column_b": "String"}, w.container_classes[1].methods[0].params)
+        self.assertEqual("This function does ratio estimation", w.container_classes[1].methods[0].docs)
+        self.assertEqual("String", w.container_classes[1].methods[0].returns)
+
+
+
+    def test_create_directories(self):
+        w = WrapperWriter(self.method_config, self.method_config)
+        w.structure_classes = self.structure_class
+        w.create_directories()
+
+        self.assertTrue(os.path.isdir(os.path.join(os.getcwd(), "./test_dir/python/path/")))
+        self.assertTrue(os.path.isdir(os.path.join(os.getcwd(), "./test_dir/scala/path/")))
+
+        shutil.rmtree("./test_dir/")
```

Comment:
If this test errors it may not remove those directories

Reply:
done